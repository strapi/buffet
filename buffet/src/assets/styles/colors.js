const colors = {
  darkBlue: '#1C5DE7',
  mediumBlue: '#005EEA',
  lightBlue: '#0097F6',
  blueTxt: '#333740',
  blueGradient:
    'linear-gradient(to right top, #306ded, #2f78f1, #3283f3, #3a8ef6, #4598f7)',
  orangeGradient: 'linear-gradient(to bottom right, #F65A1D, #F68E0E)',
  mediumGrey: '#979797',
  lightGrey: '#E3E9F3',
  lightGreyAlpha: 'rgba(0,0,0,0.1)',
  darkOrange: '#F64D0A',
  orangeBorder: 'rgba(255, 0, 0, 0.2)',
  orangeBkgd: 'rgba(255, 0, 0, 0.15)',
  black: '#000000',
  white: '#ffffff',

  // Specific to inputs
  greyPlaceholder: '#919BAE',
  greyIconBkgd: '#FAFAFB',
  greyIconColor: '#B3B5B9',
  blueBorder: '#78caff',
};
export default colors;
