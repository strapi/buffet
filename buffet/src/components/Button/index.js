/**
 *
 * Button
 *
 */

export { default } from '../../styled/Button';
