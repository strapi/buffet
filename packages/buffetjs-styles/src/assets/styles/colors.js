const colors = {
  darkBlue: '#1C5DE7',
  mediumBlue: '#005EEA',
  lightBlue: '#0097F6',
  blueTxt: '#333740',
  blueGradient:
    'linear-gradient(to right top, #306ded, #2f78f1, #3283f3, #3a8ef6, #4598f7)',
  orangeGradient: 'linear-gradient(to bottom right, #F65A1D, #F68E0E)',
  mediumGrey: '#979797',
  brightGrey: '#9ea7b8',
  lightGrey: '#E3E9F3',
  lightGreyAlpha: 'rgba(0,0,0,0.1)',
  darkOrange: '#F64D0A',
  orangeBorder: 'rgba(255, 0, 0, 0.2)',
  orangeBkgd: 'rgba(255, 0, 0, 0.15)',
  black: '#000000',
  white: '#ffffff',

  // Specific to inputs
  greyPlaceholder: '#919BAE',
  greyIconBkgd: '#FAFAFB',
  greyIconColor: '#B3B5B9',
  blueBorder: '#78caff',
  greyToggle: '#CED3DB',

  // Specific to table
  greyHeader: '#F3F3F4',
  greySubtitle: '#787E8F',
  greyHover: '#F7F8F8',
  greySeparator: '#f1f1f2',

  // Specific to timepicker
  blueIconTxt: '#367CF7',
  blueIconBkgd: '#E8F0FA',

  // Specific to tabs
  greyTabBkgd: '#F2F2F2',
  blueTabBorder: '#1C5DE7',
  greyLink: '#F5F5F5',
};
export default colors;
