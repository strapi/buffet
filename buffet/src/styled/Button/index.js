/**
 *
 * Button
 *
 */

import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';
import colors from '../../assets/styles/colors';
import sizes from '../../assets/styles/sizes';
import mixins from '../../assets/styles/mixins';
import img from '../../assets/icons/icon_trash.svg';

const Button = styled.button`

  /*
  * Theme
  */ 

  // General style
  height: ${sizes.button.height.large};
  padding: 0 ${sizes.button.padding.large};
  font-family: 'Lato';
  font-weight: 600;
  font-size: 13px;
  line-height: 100%;
  border-radius: ${sizes.radius};
  cursor:pointer;
  outline: 0;
  &:hover {
    ${mixins(colors.lightGreyAlpha).bshadow};
  }
  &:active {
    ${mixins(colors.lightGreyAlpha).bshadow};
  }

  // Specific style
  ${props =>
    props.color === 'primary' &&
    css`
      background-image: ${colors.blueGradient};
      color: ${colors.white};
      border: 0;
    `}
    
  ${props =>
    props.color === 'secondary' &&
    css`
      border: 1px solid ${colors.darkBlue};
      color: ${colors.darkBlue};
      padding: 0 ${sizes.button.padding.small};
      height: ${sizes.button.height.small};
    `}}

  ${props =>
    props.color === 'cancel' &&
    css`
      border: 1px solid ${colors.mediumGrey};
      color: ${colors.mediumGrey};
    `}}

  ${props =>
    props.color === 'delete' &&
    css`
      border: 1px solid ${colors.orangeBorder};
      background-color: ${colors.orangeBkgd};
      color: ${colors.darkOrange};
      padding: 0 ${sizes.button.padding.small};
      display: flex;
    `}


  /*
  * Left icons
  */

  // General style
  ${props =>
    !!props.icon &&
    css`
      &::before {
        font-family: 'FontAwesome';
        font-weight: 600;
        font-size: 13px;
        margin-right: 8px;
        line-height: 100%;
        display: inline-block;
        vertical-align: middle;
      }
    `}   

  // Icon specified
  ${props =>
    props.icon === 'add' &&
    css`
      &::before {
        content: '\f067';
        margin-right: 13px;
      }
    `}      

  ${props =>
    props.icon === 'back' &&
    css`
      &::before {
        content: '\f053';
      }
    `}      

  ${props =>
    props.icon === 'trash' &&
    css`
      &::before {
        content: '';
        background-image: url(${img});
        background-repeat: no-repeat;
        background-position: left;
        width: 11px;
        height: 12px;
      }
    `}
`;

Button.defaultProps = {
  color: 'primary',
  icon: null,
  type: 'button',
};

Button.propTypes = {
  color: PropTypes.oneOf(['primary', 'secondary', 'cancel', 'delete']),
  icon: PropTypes.oneOf(['add', 'back', 'trash']),
  type: PropTypes.string,
};

export default Button;
