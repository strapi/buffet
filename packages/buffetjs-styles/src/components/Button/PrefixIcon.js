import styled from 'styled-components';

import sizes from '../../assets/styles/sizes';
import Icon from '../Icon';

const PrefixIcon = styled(Icon)`
  margin-right: ${sizes.margin * 0.8}px;
  font-size: 0.94em;
`;

export default PrefixIcon;
