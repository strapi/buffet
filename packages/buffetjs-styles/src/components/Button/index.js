/**
 *
 * Button
 *
 */

import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';
import colors from '../../assets/styles/colors';
import sizes from '../../assets/styles/sizes';
import mixins from '../../assets/styles/mixins';

const Button = styled.button`

  /*
  * Theme
  */

  // General style
  height: ${sizes.button.height.large};
  padding: 0 ${sizes.button.padding.large};
  font-weight: 600;
  font-size: 1.3rem;
  line-height: normal;
  border-radius: ${sizes.radius};
  cursor: pointer;
  outline: 0;
  background-color: ${colors.white};
  &:hover {
    ${mixins(colors.lightGreyAlpha).bshadow};
  }
  &:active {
    ${mixins(colors.lightGreyAlpha).bshadow};
  }

  // Specific style
  ${props =>
    props.color === 'primary' &&
    css`
      background-image: ${colors.blueGradient};
      color: ${colors.white};
      border: 0;
    `}

  ${props =>
    props.color === 'secondary' &&
    css`
      border: 1px solid ${colors.darkBlue};
      color: ${colors.darkBlue};
      padding: 0 ${sizes.button.padding.small};
      height: ${sizes.button.height.small};
    `}}

  ${props =>
    props.color === 'cancel' &&
    css`
      border: 1px solid ${colors.mediumGrey};
      color: ${colors.mediumGrey};
    `}}

  ${props =>
    props.color === 'delete' &&
    css`
      border: 1px solid ${colors.orangeBorder};
      background-color: ${colors.orangeBkgd};
      color: ${colors.darkOrange};
      padding: 0 ${sizes.button.padding.small};
    `}
`;

Button.defaultProps = {
  color: 'primary',
  icon: false,
  type: 'button',
};

Button.propTypes = {
  color: PropTypes.oneOf(['primary', 'secondary', 'cancel', 'delete']),
  type: PropTypes.string,
};

export default Button;
