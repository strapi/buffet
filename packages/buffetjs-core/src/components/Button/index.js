/**
 *
 * Button
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { omit } from 'lodash';
import { Button as StyledButton } from '@buffetjs/styles';
import PrefixIcon from '../PrefixIcon';

function Button(props) {
  const content = props.label ? props.label : props.children;
  const icon =
    props.icon === true ? (
      <PrefixIcon icon="plus" />
    ) : (
      <PrefixIcon icon={props.icon} />
    );

  return (
    <StyledButton {...omit(props, ['icon'])}>
      {icon}
      {content}
    </StyledButton>
  );
}

Button.defaultProps = {
  children: null,
  color: 'primary',
  icon: false,
  label: null,
  type: 'button',
};

Button.propTypes = {
  children: PropTypes.node,
  color: PropTypes.string,
  icon: PropTypes.oneOfType([PropTypes.node, PropTypes.bool]),
  label: PropTypes.string,
  type: PropTypes.oneOf(['submit', 'reset', 'button', 'menu']),
};

export default Button;
