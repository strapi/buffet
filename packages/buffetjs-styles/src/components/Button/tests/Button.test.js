import React from 'react';
import { mount } from 'enzyme';
import { enzymeFind } from 'styled-components/test-utils';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Button from '../index';
import colors from '../../../assets/styles/colors';

describe('<Button />', () => {
  describe('<Button /> attributes', () => {
    it('should render an <button> tag', () => {
      const wrapper = mount(<Button />);
      const renderedComponent = enzymeFind(wrapper, Button);
      expect(renderedComponent.type()).toEqual('button');
    });

    it('should have a className attribute', () => {
      const wrapper = mount(<Button />);
      const renderedComponent = enzymeFind(wrapper, Button);
      expect(renderedComponent.prop('className')).toBeDefined();
    });

    it('should adopt a valid attribute', () => {
      const id = 'test';
      const wrapper = mount(<Button id={id} />);
      const renderedComponent = enzymeFind(wrapper, Button);
      expect(renderedComponent.prop('id')).toEqual(id);
    });

    it('should not adopt an invalid attribute', () => {
      const wrapper = mount(<Button attribute="test" />);
      const renderedComponent = enzymeFind(wrapper, Button);
      expect(renderedComponent.prop('attribute')).toBeUndefined();
    });

    it('should adopt a default type button', () => {
      const wrapper = mount(<Button />);
      const renderedComponent = enzymeFind(wrapper, Button);
      expect(renderedComponent.prop('type')).toEqual('button');
    });

    it('should adopt a type submit', () => {
      const wrapper = mount(<Button type="submit" />);
      const renderedComponent = enzymeFind(wrapper, Button);
      expect(renderedComponent.prop('type')).toEqual('submit');
    });
  });

  describe('<Button /> icons', () => {
    it('should render the add icon', () => {
      const tree = renderer.create(<Button icon="add" />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render a custom icon', () => {
      const tree = renderer.create(<Button icon={<p>Icon</p>} />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render the add trash', () => {
      const tree = renderer.create(<Button icon="trash" />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render the primary style', () => {
      const tree = renderer.create(<Button color="primary" />).toJSON();
      expect(tree).toHaveStyleRule('color', colors.white);
    });

    it('should render the secondary style', () => {
      const tree = renderer.create(<Button color="secondary" />).toJSON();
      expect(tree).toHaveStyleRule('color', colors.darkBlue);
    });

    it('should render the cancel style', () => {
      const tree = renderer.create(<Button color="cancel" />).toJSON();
      expect(tree).toHaveStyleRule('color', colors.mediumGrey);
    });

    it('should render the delete style', () => {
      const tree = renderer.create(<Button color="delete" />).toJSON();
      expect(tree).toHaveStyleRule('color', colors.darkOrange);
    });
  });
});
