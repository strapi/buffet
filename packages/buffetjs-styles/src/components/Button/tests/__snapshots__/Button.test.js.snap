// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> <Button /> icons should render a custom icon 1`] = `
.c0 {
  height: 30px;
  padding: 0 30px;
  font-weight: 600;
  font-size: 1.3rem;
  line-height: normal;
  border-radius: 3px;
  cursor: pointer;
  outline: 0;
  background-color: #ffffff;
  background-image: linear-gradient(to right top,#306ded,#2f78f1,#3283f3,#3a8ef6,#4598f7);
  color: #ffffff;
  border: 0;
}

.c0:hover {
  -webkit-box-shadow: inset 0 0 30px rgba(0,0,0,0.1);
  -moz-box-shadow: inset 0 0 30px rgba(0,0,0,0.1);
  box-shadow: inset 0 0 30px [object Object];
}

.c0:active {
  -webkit-box-shadow: inset 0 0 30px rgba(0,0,0,0.1);
  -moz-box-shadow: inset 0 0 30px rgba(0,0,0,0.1);
  box-shadow: inset 0 0 30px [object Object];
}

<button
  className="c0"
  color="primary"
  type="button"
/>
`;

exports[`<Button /> <Button /> icons should render the add icon 1`] = `
.c0 {
  height: 30px;
  padding: 0 30px;
  font-weight: 600;
  font-size: 1.3rem;
  line-height: normal;
  border-radius: 3px;
  cursor: pointer;
  outline: 0;
  background-color: #ffffff;
  background-image: linear-gradient(to right top,#306ded,#2f78f1,#3283f3,#3a8ef6,#4598f7);
  color: #ffffff;
  border: 0;
}

.c0:hover {
  -webkit-box-shadow: inset 0 0 30px rgba(0,0,0,0.1);
  -moz-box-shadow: inset 0 0 30px rgba(0,0,0,0.1);
  box-shadow: inset 0 0 30px [object Object];
}

.c0:active {
  -webkit-box-shadow: inset 0 0 30px rgba(0,0,0,0.1);
  -moz-box-shadow: inset 0 0 30px rgba(0,0,0,0.1);
  box-shadow: inset 0 0 30px [object Object];
}

<button
  className="c0"
  color="primary"
  type="button"
/>
`;

exports[`<Button /> <Button /> icons should render the add trash 1`] = `
.c0 {
  height: 30px;
  padding: 0 30px;
  font-weight: 600;
  font-size: 1.3rem;
  line-height: normal;
  border-radius: 3px;
  cursor: pointer;
  outline: 0;
  background-color: #ffffff;
  background-image: linear-gradient(to right top,#306ded,#2f78f1,#3283f3,#3a8ef6,#4598f7);
  color: #ffffff;
  border: 0;
}

.c0:hover {
  -webkit-box-shadow: inset 0 0 30px rgba(0,0,0,0.1);
  -moz-box-shadow: inset 0 0 30px rgba(0,0,0,0.1);
  box-shadow: inset 0 0 30px [object Object];
}

.c0:active {
  -webkit-box-shadow: inset 0 0 30px rgba(0,0,0,0.1);
  -moz-box-shadow: inset 0 0 30px rgba(0,0,0,0.1);
  box-shadow: inset 0 0 30px [object Object];
}

<button
  className="c0"
  color="primary"
  type="button"
/>
`;
